File: GoLang/8 kyu/To square (root) or not to square (root)/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(SquareOrSquareRoot([]int{10, 9, 37, 36, 1}))
}

func SquareOrSquareRoot(arr []int) []int{
  var res []int
  for _, value := range arr {
    
    sqrt := math.Sqrt(float64(value))
    
    if sqrt == math.Floor(sqrt) {
      res = append(res, int(sqrt))
    } else {
      res = append(res, value*value)
    }
    
  }
  
  return res
}

File: Python/8 kyu/To square (root) or not to square (root)/main.py
import math

def square_or_square_root(arr):
    for i in range(len(arr)):
        sqrt = math.sqrt(arr[i])
        if sqrt == int(sqrt):
            arr[i] = int(sqrt)
        else:
            arr[i] *= arr[i]
    return arr

if __name__ == "__main__":
    print(square_or_square_root([10, 9, 37, 36, 1]))